Ginger/Ginger/UserControlsLib/ActionInputValueUserControlLib/ActionInputValueUserControl.xaml.cs

```diff
@@ -36,17 +40,136 @@ public partial class ActionInputValueUserControl : UserControl
     {
         ActInputValue mActInputValue;
         Context mContext;
-
-        public ActionInputValueUserControl(Context context, ActInputValue actInputValue)
+        List<Attribute> mActionParamProperties;
+        string mLabel;
+        object mDefaultValue;
+        bool mBrowseNeeded = false;
+        Environment.SpecialFolder mFolderType = Environment.SpecialFolder.MyComputer;
+        Ginger.Activities.UCValueExpression.eBrowserType mBrowseType = Ginger.Activities.UCValueExpression.eBrowserType.File;
+        string mFileType = string.Empty;
+        public ActionInputValueUserControl(Context context, ActInputValue actInputValue, List<Attribute> actionParamProperties)
         {
             InitializeComponent();
 
             mActInputValue = actInputValue;
             mContext = context;
+            mActionParamProperties = actionParamProperties;
 
             ResetControls();
 
+            mLabel = char.ToUpper(mActInputValue.Param[0]) +  mActInputValue.Param.Substring(1);   // capitilize first letter only of the param name
+
+            SetParamLayout();
             SetControlToInputValue();
+            SetParamValidations();
+        }
+
+        private void SetParamValidations()
+        {
+            if (mActionParamProperties == null)
+            {
+                return;
+            }
+
+            foreach (Attribute attr in mActionParamProperties)
+            {             
+                if (attr.GetType() == typeof(MandatoryAttribute))
```

Comment:
Using reflection get all classes with interface IActionParamProperty - then loop on them instead all the if/else below

Reply:
even if we used reflection to get all classes, we still need to check type of class because validation rule is different for each attribute.